File: PrezBDDTDD.Tests/Steps/GestionDesVirementsGivenSteps.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using PrezBDDTDD.Tests.Model;
using PrezBDDTDD.Tests.Process;
using System;
using TechTalk.SpecFlow;

namespace PrezBDDTDD.Tests.Steps
{
    [Binding]
    public class GestionDesVirementsGivenSteps
    {
        public GestionDesVirementsGivenSteps(Context ctx)
        {
            this.Context = ctx;
        }

        public Context Context { get; private set; }

        [Given(@"j'ai un compte cheque avec un solde de (.*)€")]
        public void GivenJAiUnCompteChequeAvecUnSoldeDe(int solde)
        {
            Context.CompteCheque.Solde = solde;
        }
        
        [Given(@"j'ai un compte épargne avec un solde de (.*)€")]
        public void GivenJAiUnCompteEpargneAvecUnSoldeDe(int solde)
        {
            Context.CompteEpargne.Solde = solde;
        }
        
        [Given(@"la limite de virement est (.*)€")]
        public void GivenLaLimiteDeVirementEst(int limite)
        {
            Context.srvVirement.Plafond = limite;
        }        
    }
}

File: PresBDDTDD.Tests.IOSSwift/PrezBDDTDD/PrezBDDTDDCucumberTests/Step Definitions/GestionDesVirementsGivenSteps.swift
//
//  GestionDesVirementsGivenSteps.swift
//  PrezBDDTDD
//
//  Created by Yann Breleur on 14/02/2017.
//  Copyright © 2017 AXA. All rights reserved.
//

import Foundation
import XCTest
import Cucumberish

class GestionDesVirementsGivenSteps: NSObject {
    
    var context:ContextGestionVirement
    
    init(_ context: ContextGestionVirement) {
        self.context = context
    }
    
    func GestionDesVirementsGivenSteps() {
        
        Given("compte cheque avec un solde de ([0-9]*)") { (args, userInfo) -> Void in
            let solde = Int((args?[0]) ?? "0") ?? 0
            self.context.compteCheque.solde = solde
        }
        
        Given("j'ai un compte épargne avec un solde de ([0-9]*)€"){ (args, userInfo) -> Void in
            let solde = Int((args?[0]) ?? "0") ?? 0
            self.context.compteEpargne.solde = solde
        }
        
        Given("la limite de virement est ([0-9]*)€"){ (args, userInfo) -> Void in
            let limiteVirement = Int((args?[0]) ?? "0") ?? 0
            self.context.limiteVirement = limiteVirement
        }
        
    }
}
